worker/src/RTC/IceServer.cpp

```diff
@@ -300,27 +312,136 @@ namespace RTC
 
 			case RTC::StunPacket::Class::INDICATION:
 			{
-				MS_DEBUG_TAG(ice, "STUN Binding Indication processed");
+				MS_DEBUG_TAG(ice, "STUN Binding Indication received, ignored");
 
 				break;
 			}
 
 			case RTC::StunPacket::Class::SUCCESS_RESPONSE:
 			{
-				MS_DEBUG_TAG(ice, "STUN Binding Success Response processed");
+				// Check authentication.
+				switch (packet->CheckAuthentication(
+				  this->remoteUsernameFragment, this->usernameFragment, this->remotePassword))
+				{
+					case RTC::StunPacket::Authentication::OK:
+					{
+						break;
+					}
+
+					case RTC::StunPacket::Authentication::UNAUTHORIZED:
+					{
+						MS_WARN_TAG(ice, "received STUN Success Response with wrong authentication, discarded");
+
+						return;
+					}
+
+					case RTC::StunPacket::Authentication::BAD_MESSAGE:
+					{
+						MS_WARN_TAG(
+						  ice,
+						  "cannot check authentication in received STUN Binding Success Response, discarded");
+
+						return;
+					}
+				}
+
+				// Trick: We only read the fist 4 bytes of the transactionId of the
+				// response since we know that we generated 4 bytes followed by 8 zeroed
+				// bytes in the transactionId of the sent consent request.
+				auto transactionId = Utils::Byte::Get4Bytes(packet->GetTransactionId(), 0);
+
+				// Let's iterate all entries in the queue and remove the consent request
+				// associated to this response and all those that were sent before.
+
+				// Find the associated entry first.
+				auto it = this->sentConsents.begin();
+
+				for (; it != this->sentConsents.end(); ++it)
+				{
+					auto& sentConsent = *it;
+
+					if (Utils::Byte::Get4Bytes(sentConsent.transactionId, 0) == transactionId)
+					{
+						break;
+					}
+				}
```

Comment:
We don't use `find_if` anywhere. It may look more idiomatic but it requires passing a callback, which means a separate function declared somewhere in the file. How is that better? Or do I miss something?